yt_dlp/utils/_utils.py

```diff
@@ -3234,7 +3234,15 @@ def fix_kv(m):
     def create_map(mobj):
         return json.dumps(dict(json.loads(js_to_json(mobj.group(1) or '[]', vars=vars))))
 
+    def create_array(mobj):
+        return mobj.group(1) + js_to_json(f'[{mobj.group(2)}]', vars=vars) + mobj.group(3)
+
     code = re.sub(r'new Map\((\[.*?\])?\)', create_map, code)
+    while True:
+        old_code = code
+        code = re.sub(r'^(.*?)(?:new\s+)?Array\((.*?)\)(.*?)$', create_array, old_code)
+        if old_code == code:
+            break
```

Comment:
```suggestion
    code = re.sub(r'(?:new\s+)?Array\((.*?)\)', r'[\g<1>]', code)
    code = re.sub(r'new Map\((\[.*?\])?\)', create_map, code)
```
I think this should solve the same thing. Since `Map` uses arrays, we would want to replace with the literal ayntax before doing that substitution.

Since this could cause regressions I would add tests for this in `test_utils.py`. Alternatively we can hotfix it only in the extractor, by doing the `html = re.sub(r'(?:new\s+)?Array\((.*?)\)', r'[\g<1>]', html)` replacement beforehand so we can be sure it wont affect other extractors. Make sure to check if my suggestion works